File: php/microtime.php
<?php
if ($argc != 2) {
    echo "Usage: php " . $argv[0] . " <iterations>\n";
    exit(1);
}

$iterations = (int)$argv[1];

$start = microtime(true);

for ($i = 0; $i < $iterations; $i++) {
    microtime(true);
}

$end = microtime(true);
echo sprintf('%.12f', $end - $start) . "\n";

File: c/microtime.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/time.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Usage: %s <iterations>\n", argv[0]);
        return 1;
    }

    int iterations = atoi(argv[1]);

    struct timeval start, end;
    gettimeofday(&start, NULL);

    for (int i = 0; i < iterations; i++) {
        struct timeval tv;
        gettimeofday(&tv, NULL);
    }

    gettimeofday(&end, NULL);

    long seconds = (end.tv_sec - start.tv_sec);
    long micros = ((seconds * 1000000) + end.tv_usec) - (start.tv_usec);

    printf("%f\n", (double)micros / 1000000);

    return 0;
}
